internal/civisibility/integrations/gotesting/testcontroller_test.go

```diff
@@ -65,34 +66,32 @@ func TestMain(m *testing.M) {
 	} else if internal.BoolEnv("Bypass", false) {
 		os.Exit(m.Run())
 	} else {
-		fmt.Println("Starting tests...")
-		var wg sync.WaitGroup
 		for _, v := range scenarios {
-			wg.Add(1)
 			cmd := exec.Command(os.Args[0], os.Args[1:]...)
-			cmd.Stdout, cmd.Stderr = os.Stdout, os.Stderr
+			var b bytes.Buffer
+			if log.DebugEnabled() {
+				cmd.Stdout, cmd.Stderr = os.Stdout, os.Stderr
+			} else {
+				cmd.Stdout = &b
+				cmd.Stderr = &b
+			}
 			cmd.Env = append(cmd.Env, os.Environ()...)
 			cmd.Env = append(cmd.Env, fmt.Sprintf("%s=true", v))
-			go func() {
-				defer func() {
-					wg.Done()
-				}()
-
-				fmt.Printf("Running scenario: %s:\n", v)
-				fmt.Println(cmd.Env)
-				err := cmd.Run()
-				fmt.Printf("Done.\n\n")
-				if err != nil {
-					if exiterr, ok := err.(*exec.ExitError); ok {
-						fmt.Printf("Scenario %s failed with exit code: %d\n", v, exiterr.ExitCode())
-						os.Exit(exiterr.ExitCode())
+			fmt.Printf("\n**** [RUNNING SCENARIO: %s]\n", v)
+			err := cmd.Run()
+			fmt.Printf("\n**** [SCENARIO %s IS DONE]\n\n", v)
+			if err != nil {
+				if exiterr, ok := err.(*exec.ExitError); ok {
+					fmt.Printf("\n===========================================\n**** [SCENARIO %s FAILED WITH EXIT CODE: %d]\n", v, exiterr.ExitCode())
```

Comment:
Thank you for this! One comment: any reason why we're using `fmt.Printf` here but `log.Debug` below? We also have the `log.Info` function if we need to print regardless of debug mode.